File: C#/WindowsPhone/AVGSample/AVGSample/MyAVGScreen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Loon.Core.Graphics.Component;
using Loon.Action.Avg;
using Loon.Core.Graphics.Opengl;
using Loon.Core.Graphics;
using Loon.Action.Avg.Drama;

namespace AVGTest
{
    public class MyAVGScreen : AVGScreen
    {

        LPaper roleName;

        String flag = "自定义命令.";

        String[] selects = { "鹏凌三千帅不帅？" };

        int type;

        public MyAVGScreen()
            : base("assets/script/s1.txt", AVGDialog.GetRMXPDialog("assets/w6.png",
                460, 150))
        {

        }

        public override void OnLoading()
        {
            roleName = new LPaper("assets/name0.png", 25, 25);
            LeftOn(roleName);
            roleName.SetLocation(5, 15);
            Add(roleName);
        }

        public override void DrawScreen(GLEx g)
        {
            switch (type)
            {
                case 1:
                    g.SetAntiAlias(true);
                    g.DrawSixStart(LColor.yellow, 130, 100, 100);
                    g.SetAntiAlias(false);
                    break;
            }
            g.ResetColor();
        }

        LButton yes;

        public override void InitCommandConfig(Command command)
        {
            // 初始化时预设变量
            command.SetVariable("p", "assets/p.png");
            command.SetVariable("sel0", selects[0]);
        }

        public override void InitMessageConfig(LMessage message)
        {

        }

        public override void InitSelectConfig(LSelect select)
        {
        }

        class YesClick : ClickListener
        {
            public void DoClick(LComponent comp)
            {
             
            }

            public void DownClick(LComponent comp, float x, float y)
            {

                if (comp.Tag is AVGScreen)
                {
                    AVGScreen screen = (AVGScreen)comp.Tag;
                    // 解除锁定
                    screen.SetLocked(false);
                    // 触发事件
                    // click();
                    // 删除当前按钮
                    screen.Remove(comp);
                }
            }

            public void UpClick(LComponent comp, float x, float y)
            {

            }

            public void DragClick(LComponent comp, float x, float y)
            {

            }
        }

        public override bool NextScript(String mes)
        {

            // 自定义命令（有些自定义命令为了突出写成了中文，实际不推荐）
            if (roleName != null)
            {
                if ("noname".Equals(mes, StringComparison.InvariantCultureIgnoreCase))
                {
                    roleName.SetVisible(false);
                }
                else if ("name0".Equals(mes, StringComparison.InvariantCultureIgnoreCase))
                {
                    roleName.SetVisible(true);
                    roleName.SetBackground("assets/name0.png");
                    roleName.SetLocation(5, 15);
                }
                else if ("name1".Equals(mes, StringComparison.InvariantCultureIgnoreCase))
                {
                    roleName.SetVisible(true);
                    roleName.SetBackground("assets/name1.png");
                    roleName.SetLocation(GetWidth() - roleName.GetWidth() - 5, 15);
                }
            }
            if ((flag + "星星").Equals(mes, StringComparison.InvariantCultureIgnoreCase))
            {
                // 添加脚本事件标记（需要点击后执行）
                SetScrFlag(true);
                type = 1;
                return false;
            }
            else if ((flag + "去死吧，星星").Equals(mes, StringComparison.InvariantCultureIgnoreCase))
            {
                type = 0;
            }
            else if ((flag + "关于天才").Equals(mes, StringComparison.InvariantCultureIgnoreCase))
            {
                message.SetVisible(false);
                SetScrFlag(true);
                // 强行锁定脚本
                SetLocked(true);
                yes = new LButton("assets/dialog_yes.png", 112, 33);
                yes.Tag = this;
                yes.Click = new YesClick();
                CenterOn(yes);
                Add(yes);
                return false;
            }
            return true;
        }

        public override void OnExit()
        {
            // 重新返回标题画面
            SetScreen(new AVGTitle());
        }

        public override void OnSelect(String message, int type)
        {
            if (selects[0].Equals(message, StringComparison.InvariantCultureIgnoreCase))
            {
                command.SetVariable("sel0", Convert.ToString(type));
            }
        }

    }
}

File: Java/Loon-Neo-GWT/src/loon/test/MyAVGScreen.java
package loon.test;

import loon.action.avg.AVGDialog;
import loon.action.avg.AVGScreen;
import loon.action.avg.drama.Command;
import loon.canvas.LColor;
import loon.component.LButton;
import loon.component.LMessage;
import loon.component.LPaper;
import loon.component.LSelect;
import loon.events.GameTouch;
import loon.opengl.GLEx;

public class MyAVGScreen extends AVGScreen {

	LPaper roleName;

	// 自定义命令（有些自定义命令为了突出写成了中文，实际不推荐）
	String flag = "自定义命令.";

	String[] selects = { "鹏凌三千帅不帅？" };

	int type;

	public MyAVGScreen() {
		super("assets/script/s1.txt", AVGDialog.shared().getRMXPDialog("assets/w6.png",
				460, 150));
	}

	public void onLoading() {
		roleName = new LPaper("assets/name0.png", 25, 25);
		leftOn(roleName);
		roleName.setLocation(5, 15);
		add(roleName);
	}

	public void drawScreen(GLEx g) {
		switch (type) {
		case 1:
			g.drawSixStart(LColor.yellow, 130, 100, 100);
			break;
		}
		g.resetColor();
	}

	public void initCommandConfig(Command command) {
		// 初始化时预设变量
		command.setVariable("p", "assets/p.png");
		command.setVariable("sel0", selects[0]);
	}

	public void initMessageConfig(LMessage message) {

	}

	public void initSelectConfig(LSelect select) {
	}

	public boolean nextScript(String mes) {

		// 自定义命令（有些自定义命令为了突出写成了中文，实际不推荐）
		if (roleName != null) {
			if ("noname".equalsIgnoreCase(mes)) {
				roleName.setVisible(false);
			} else if ("name0".equalsIgnoreCase(mes)) {
				roleName.setVisible(true);
				roleName.setBackground("assets/name0.png");
				roleName.setLocation(5, 15);
			} else if ("name1".equalsIgnoreCase(mes)) {
				roleName.setVisible(true);
				roleName.setBackground("assets/name1.png");
				roleName.setLocation(getWidth() - roleName.getWidth() - 5, 15);
			}
		}
		if ((flag + "星星").equalsIgnoreCase(mes)) {
			// 添加脚本事件标记（需要点击后执行）
			setScrFlag(true);
			type = 1;
			return false;
		} else if ((flag + "去死吧，星星").equalsIgnoreCase(mes)) {
			type = 0;
		} else if ((flag + "关于天才").equalsIgnoreCase(mes)) {
			messageUI.setVisible(false);
			setScrFlag(true);
			// 强行锁定脚本
			setScriptLocked(true);
			LButton yes = new LButton("assets/dialog_yes.png", 112, 33) {
				public void doClick() {
					// 解除锁定
					setScriptLocked(false);
					// 触发事件
					// click();
					// 删除当前按钮
					remove(this);
				}
			};
			centerOn(yes);
			add(yes);
			return false;
		}
		return true;
	}

	public void onExit() {
		// 重新返回标题画面
		setScreen(new TitleScreen());
	}

	public void onSelect(String message, int type) {
		if (selects[0].equalsIgnoreCase(message)) {
			command.setVariable("sel0", String.valueOf(type));
		}
	}

	@Override
	public void resize(int width, int height) {
		// TODO Auto-generated method stub

	}

	@Override
	public void touchDrag(GameTouch e) {
		// TODO Auto-generated method stub

	}

	@Override
	public void resume() {
		// TODO Auto-generated method stub

	}

	@Override
	public void pause() {
		// TODO Auto-generated method stub

	}

}
